From 078ebc5a64cd42adbe5a845c8cb43a5d75b84af6 Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Fri, 14 Sep 2012 23:07:01 +0200
Subject: [PATCH 41/82] Fix connection id setting

Set connection id to AR5523_ID_BSS then connected. Not the other way around.
This should fix TX speed regression.
---
 ar5523.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/wireless/ath/ar5523/ar5523.c
+++ b/drivers/net/wireless/ath/ar5523/ar5523.c
@@ -976,9 +976,9 @@ static void ar5523_tx_work(struct work_s
 		desc->flags  = 0;
 
 		if (test_bit(AR5523_CONNECTED, &ar->flags))
-			desc->connid  = cpu_to_be32(AR5523_ID_BROADCAST);
+			desc->connid = cpu_to_be32(AR5523_ID_BSS);
 		else
-			desc->connid  = cpu_to_be32(AR5523_ID_BSS);
+			desc->connid = cpu_to_be32(AR5523_ID_BROADCAST);
 
 		if (txi->flags & IEEE80211_TX_CTL_USE_MINRATE)
 			txqid |= UATH_TXQID_MINRATE;
